### PR TITLE
Timestamp fixes, uuid support

### LINKED ISSUE #48: java.time.Instant stored without nanoseconds
In `Session.kt` in `fun PreparedStatement.setParam(idx: Int, v: Any?)` we have
```
is Instant -> this.setTimestamp(idx, Timestamp(Date.from(v).time))
```

In result `Instant` is converted to `Date`, and than to `Timestamp`

`Instant` and `Timestamp` support nanoseconds, 
But `Date` does not. So value is truncated.
We can map value directly using `Timestamp.from(Instant)`

Another issue in same function
```
when (v) {
  ...
  is java.util.Date -> this.setTimestamp(idx, Timestamp(v.time))
  is java.sql.Timestamp -> this.setTimestamp(idx, v)
  ...
```
`Timestamp` in sub class of `Date`. So `Timestamp` is not executed. And we loose nanos here as well 

P.S. Thanks for great lib!

### PR DESCRIPTION
**Pull request by ievgen-kapinos**
Closes #48 
Plus:
* Added tests
* Added UUID support
* Improved existed tests
* Added ability to configure Hikari
* Deprecated `HikariCP.default(...)` because redundant
* Updated dependency versions
* Applied same style for not imported classes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -30,13 +30,14 @@ repositories {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    implementation 'org.slf4j:slf4j-api:1.7.32'
-    implementation 'joda-time:joda-time:2.10.10'
+    implementation "org.slf4j:slf4j-api:1.7.36"
+    implementation "joda-time:joda-time:2.10.13"
     // TODO: upgrade to 5+ when dropping Java 8 supports
-    implementation 'com.zaxxer:HikariCP:4.0.3'
+    implementation "com.zaxxer:HikariCP:5.0.1"
```
Comment (seratch) [inline build.gradle]: Can you revert this major upgrade this time? We'll do this in the future but not now.

File: src/main/kotlin/kotliquery/HikariCP.kt [hunk 2]
```diff
@@ -9,18 +9,36 @@ object HikariCP {
 
     private val pools: ConcurrentMap<String, HikariDataSource> = ConcurrentHashMap()
 
-    fun default(url: String, username: String, password: String): HikariDataSource {
+    @Deprecated(
```
Comment (seratch) [inline src/main/kotlin/kotliquery/HikariCP.kt]: Can you revert this? I'm not planning to make this deprecated.

File: src/main/kotlin/kotliquery/HikariCP.kt [hunk 3]
```diff
@@ -9,18 +9,36 @@ object HikariCP {
...
     }
 
-    fun init(name: String, url: String, username: String, password: String): HikariDataSource {
-        val config: HikariConfig = HikariConfig()
+    fun init(
+        name: String = "default",
+        url: String,
+        username: String,
+        password: String,
+        configPropertiesFn: HikariConfig.() -> Unit = { }
```
Comment (seratch) [inline src/main/kotlin/kotliquery/HikariCP.kt:21]: Nice enhancement!

File: src/main/kotlin/kotliquery/Session.kt [hunk 4]
```diff
@@ -19,7 +27,7 @@ open class Session(
     open val autoGeneratedKeys: List<String> = listOf(),
     var transactional: Boolean = false,
     open val strict: Boolean = false
-) : AutoCloseable {
+) : Closeable {
```
Comment (seratch) [inline src/main/kotlin/kotliquery/Session.kt:30]: This is fine but can you share the reason why I've changed this interface to sub interface?

File: src/main/kotlin/kotliquery/Session.kt [hunk 5]
```diff
@@ -49,41 +57,33 @@ open class Session(
...
+                is LocalDateTime -> this.setTimestamp(idx, java.sql.Timestamp.valueOf(v))
+                is LocalDate -> this.setDate(idx, java.sql.Date.valueOf(v))
+                is LocalTime -> this.setTime(idx, java.sql.Time.valueOf(v))
+                is org.joda.time.DateTime -> this.setTimestamp(idx, java.sql.Timestamp(v.millis))
+                is org.joda.time.LocalDateTime -> this.setTimestamp(idx, java.sql.Timestamp(v.toDateTime().millis))
+                is org.joda.time.LocalDate -> this.setDate(idx, java.sql.Date(v.toDateTimeAtStartOfDay().millis))
                 is org.joda.time.LocalTime -> this.setTime(idx, java.sql.Time(v.toDateTimeToday().millis))
-                is java.util.Date -> this.setTimestamp(idx, Timestamp(v.time))
-                is java.sql.Timestamp -> this.setTimestamp(idx, v)
+                is java.sql.Timestamp -> this.setTimestamp(idx, v) // extends java.util.Date
```
Comment (seratch) [inline src/main/kotlin/kotliquery/Session.kt:70]: Thanks for fixing the order here

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for working on this! Overall, the changes look great but can you check my comments?

File: src/main/kotlin/kotliquery/HikariCP.kt (hunk 2 above)
Comment (ievgen-kapinos) [reply src/main/kotlin/kotliquery/HikariCP.kt:12]: I've realized that `init` method can do the same after assigning default value to `name` parameter

```
HikariCP.init("jdbc:h2:mem:default","user", "pass")
```

So `default(...)` method looks redundant.

File: src/main/kotlin/kotliquery/Session.kt (hunk 4 above)
Comment (ievgen-kapinos) [reply src/main/kotlin/kotliquery/Session.kt:30]: Allows to utilize `use` function build-in Kotlin

https://stackoverflow.com/a/46843631/5740547

If you take a look in its source code it is way reacher  than `LoanPattern`. Check out the source code, not a copy-paste in stackoverflow.

**Review by seratch — APPROVED** [review]
Thanks, LGTM. Will release a new version shortly

**Comment by ievgen-kapinos** [comment]
> Thanks, LGTM. Will release a new version shortly

Recently I've revisited Spring's JdbcTemplate. So much history in it's implementation, so many layers, so deep stacktraces Your lib is totally different. Well done! 

Waiting for new version! Have a nice day!

**Comment by busches** [comment]
On moving from 1.6.1 to 1.6.2 we now get the following:
```
org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid = character varying
      Hint: No operator matches the given name and argument types. You might need to add explicit type casts.
      Position: 773
        at app//org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
        at app//org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2365)
        at app//org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:355)
        at app//org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:490)
        at app//org.postgresql.jdbc.PgStatement.execute(PgStatement.java:408)
        at app//org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:166)
        at app//org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:118)
```

The query is using a `UUID` type variable into a `WHERE` clause. I would not have expected a patch release to result in a breaking change.

**Comment by ievgen-kapinos** [comment]
@busches Confirm it. I'll make a fix soon

**Pull request merged**